__integrations__/index.test.ts

```diff
@@ -0,0 +1,406 @@
+
+beforeAll(async () => {});
+afterAll(async () => {});
+
+beforeEach(() => jest.clearAllMocks());
+
+// ----------------
+
+export const aLegacyInboxEnabledFiscalCode = "AAABBB01C02D345L" as FiscalCode;
+export const aLegacyInboxDisabledFiscalCode = "AAABBB01C02D345I" as FiscalCode;
+export const anAutoFiscalCode = "AAABBB01C02D345A" as FiscalCode;
+export const aManualFiscalCode = "AAABBB01C02D345M" as FiscalCode;
+
+export const anEnabledServiceId = "anEnabledServiceId" as NonEmptyString;
+export const aDisabledServiceId = "aDisabledServiceId" as NonEmptyString;
+const aNonExistingServiceId = "aNonExistingServiceId" as NonEmptyString;
+
+// ----------------
+
+export const aMessageBodyMarkdown = "test".repeat(80);
+export const aMessageContent: MessageContent = {
+  markdown: aMessageBodyMarkdown,
+  subject: "test".repeat(10)
+};
+
+// Must correspond to an existing serviceId within "services" colletion
+const aSubscriptionKey = "aSubscriptionKey";
+
+const customHeaders = {
+  "x-user-groups":
+    "ApiUserAdmin,ApiLimitedProfileRead,ApiFullProfileRead,ApiProfileWrite,ApiDevelopmentProfileWrite,ApiServiceRead,ApiServiceList,ApiServiceWrite,ApiPublicServiceRead,ApiPublicServiceList,ApiServiceByRecipientQuery,ApiMessageRead,ApiMessageWrite,ApiMessageWriteDefaultAddress,ApiMessageList,ApiSubscriptionsFeedRead,ApiInfoRead,ApiDebugRead,ApiMessageWriteEUCovidCert",
+  "x-subscription-id": anEnabledServiceId,
+  "x-user-email": "unused@example.com",
+  "x-user-id": "unused",
+  "x-user-note": "unused",
+  "x-functions-key": "unused",
+  "x-forwarded-for": "0.0.0.0",
+  "Ocp-Apim-Subscription-Key": aSubscriptionKey
+};
+
+const mockGetCustomHeaders = jest.fn(() => customHeaders);
+
+const mockNodeFetch = jest.fn(
+  async (input: RequestInfo, init?: RequestInit) => {
+    if (SHOW_LOGS) {
+      console.log("Sending request");
+      console.log(input);
+    }
+
+    const res = await ((nodeFetch as unknown) as typeof fetch)(input, {
+      ...init,
+      headers: {
+        ...(init?.headers ?? {}),
+        ...mockGetCustomHeaders()
+      }
+    });
+
+    if (SHOW_LOGS) {
+      console.log("Result: ");
+      console.log(res);
+    }
+
+    return res;
+  }
+);
+
+const delay = ms => new Promise(resolve => setTimeout(resolve, ms));
+
+const client = createClient<"SubscriptionKey">({
+  basePath: "/api/v1",
+  baseUrl,
+  fetchApi: mockNodeFetch,
+  withDefaults: op => params =>
+    op({
+      ...params,
+      SubscriptionKey: aSubscriptionKey
+    })
+});
+
+// Wait some time
+beforeAll(async () => {
+  let i = 0;
+  while (i < 100) {
+    console.log("Waiting the function to setup..");
+    try {
+      const response = await nodeFetch("http://function:7071/api/info");
+      break;
+    } catch (e) {
+      await delay(WAIT_MS);
+      i++;
+    }
+  }
+  if (i >= 10) {
+    console.log("Function unable to setup in time");
+    exit(1);
+  }
+});
+
+describe("fn-services |> ping", () => {
+  it("/info return 200", async () => {
```

Comment:
Test is against 200 and we expect 500?

Reply:
Thanks, I remove it, since it was not necessary